pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -305,13 +308,22 @@ func NewBackupObject(cluster *v3.Cluster, manual bool) *v3.EtcdBackup {
 	}
 }
 
-func getBackupFilename(snapshotName string, cluster *v3.Cluster) string {
+func getBackupURL(snapshotName string, cluster *v3.Cluster) string {
 	if cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig == nil ||
 		cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig == nil {
 		return ""
 	}
 	target := cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig
-	return fmt.Sprintf("https://s3.%s.amazonaws.com/%s/%s", target.Region, target.BucketName, snapshotName)
+	return fmt.Sprintf("https://%s/%s/%s.%s", target.Endpoint, target.BucketName, snapshotName, compressedExtension)
+}
+
+func getBackupFilenameFromURL(URL string) (string, error) {
+	parsedURL, err := url.Parse(URL)
+	if err != nil {
+		return "", err
+	}
+	fileName := path.Base(parsedURL.Path)
```

Comment:
do we need to validate if Path is empty, in that case fileName would be `.`?

Reply:
Added isValidURL check and empty Path check